code/search/binary_search/binary_search_2.cpp

```diff
@@ -0,0 +1,64 @@
+/* Part of Cosmos by OpenGenus Foundation */
+
+#include <vector>
+#include <cstdlib>
+#include <iostream>
+#include <algorithm>
+
+using namespace std;
+
+/* Utility functions */
+
+void fill(vector<int> &v) {
+    for(int i=0;i<v.size();i++) {
+        v[i]=rand() % 100;
+    }
+}
+
+void printarr(vector<int> &v) {
+    for(int i=0;i<v.size();i++) {
+        cout << v[i] << " ";
+    }
+    cout << endl;
+}
+
+/* Binary search with fewer comparisons */
+int binary_search(vector<int> &v, int key) {
+    int m;
+    
+    int l=0,r=v.size();
+
+    while(r-l >1) {
+        m=l+(r-l)/2;
+
+        if( v[m] > key) 
+            r=m;
+        else 
+            l=m;
+    }
+    
+    if ( v[l] == key) 
+        return l;
+    else 
+        return -1;
+}
+
+/* Driver program */
+int main() {
+    int size;
+    cout << "Set array size:";
+    cin >> size;
+    
+    vector<int> v(size);
+    fill(v);
+
+    cout << "Array (sorted) : ";
+    sort(v.begin(),v.end());    
+    printarr(v);
+    
+    cout << "Search for (input search key) : ";
+    int key;
+    cin >> key;   
+    
+    cout << "Found " << key << " at index "<< binary_search(v,key) << endl;
+}
```

Comment:
Please add a new line at the last ! 👍

Reply:
What exactly do you mean by that? Just adding an empty line at the end?

Reply:
@yatharthshahjpr There already is a newline, otherwise Github would say so.